### ISSUE TITLE
Question about audio preprocessing

### ISSUE THREAD

**Issue by luckygeko** [DianJin-HFUT/SimToken#4]
Hi!! thank you for releasing this great work and the codebase.

While running training, I encountered a shape mismatch issue related to the audio features.
In the audio preprocessing stage, the extracted audio feature appears to have variable temporal length T across samples. 
However, during training, the model performs
`audio_embeddings = self.audio_feature_layer(torch.stack(audio_features, dim=0))`
This causes a batch-wise shape mismatch error when different samples have different T values (e.g., [20, 128] vs [10, 128]), since torch.stack requires equal-sized tensors.

I was wondering:
- Did you apply any fixed-length processing (e.g., temporal sampling, truncation, padding, interpolation, or pooling) to ensure a consistent T across samples?
- If so, could you clarify where this step is implemented?
- Or is the dataset expected to already have a fixed temporal length for all audio segments?

In the provided code, I could not find explicit temporal normalization for audio features, so I wanted to confirm the intended preprocessing pipeline.

Thank you very much for your help!

**Comment by DianJin-HFUT** [comment]
First, I apologize for the delayed response, and I truly appreciate your patience in waiting for an answer.

Regarding the issue you're encountering with the audio features, the dataset's audio features should indeed have a consistent shape of [10, 128]. However, due to the dual-channel nature of some of the audio samples in the dataset, and possibly due to dependencies like Towhee, some users have reported obtaining audio features with a shape of [20, 128].

To address this, we have updated the feature extraction pipeline to include a step where only the mono-channel audio is retained. This ensures that the resulting features will consistently have the shape [10, 128] across all samples.

We have validated this update, and I can confirm that it does not impact the performance of the model in terms of validation results.

Additionally, if you'd prefer, you could try using alternative methods to deploy VGGish and extract audio features, such as using torchaudio or loading the VGGish model locally.

I hope this helps clarify the issue, and thank you again for reaching out!
